### PR TITLE
vimPlugins.neuron-vim: ihsanturk/neuron.vim ->  fiatjaf/neuron.vim 

### PR DESCRIPTION
**Pull request by wbadart**
###### Motivation for this change

The current derivation for `vimPlugins.neuron-vim` points to [ihsanturk/neuron.vim], which doesn't seem to be actively maintained. This PR switches over to an actively maintained fork, [fiatjaf/neuron.vim], which works with recent versions of [neuron].

[ihsanturk/neuron.vim]: https://github.com/ihsanturk/neuron.vim
[fiatjaf/neuron.vim]: https://github.com/fiatjaf/neuron.vim
[neuron]: https://github.com/srid/neuron

###### Things done

First I removed the original ihsanturk/neuron.vim from `vim-plugin-names`. Then I ran the update script like so:

```sh
./update.py --add fiatjaf/neuron.vim
```

---

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/vim-plugins/vim-plugin-names [hunk 1]
```diff
@@ -157,7 +157,6 @@ iamcco/coc-vimlsp
 ianks/vim-tsx
 idanarye/vim-merginal
 idris-hackers/idris-vim
-ihsanturk/neuron.vim
```
Comment (SuperSandro2000) [inline pkgs/misc/vim-plugins/vim-plugin-names:160]: This is unrelated to this commit, right?

Comment (wbadart) [reply pkgs/misc/vim-plugins/vim-plugin-names:160]: That's the original plugin. I figured since it's no longer maintained, we could take it out and replace it with the new one. Not sure how to resolve the naming conflict if we should actually be keeping both.

Comment (SuperSandro2000) [reply pkgs/misc/vim-plugins/vim-plugin-names:160]: I totally agree with your change but you changed it in [`a79c400` (#101385)](https://github.com/NixOS/nixpkgs/pull/101385/commits/a79c4000a51f3b104e0aacd2b27a4cd03e2bd88b) which you named ``vimPlugins: resolve github repository redirects``. I think the change is wrong in this particular commit and maybe needs its own commit or fits into some other commit better.

Comment (wbadart) [reply pkgs/misc/vim-plugins/vim-plugin-names:160]: Gotcha. All the commits were created by the `update.py` script. I'll go through with a rebase and organize the changes a little better.

**Comment by wbadart** [comment]
hey @SuperSandro2000, I think you'll find that the commits are a little better organized now. The first one, 22d65ff, just contains the results of `update.py` with no modifications to the plugin list. The next one, 927c644, is the result of `update.py --add fiatjaf/neuron.vim` and only includes the change to the generated nix expression and `vim-plugin-names` list. Finally, 4d69f745fe96cd19a672fa991a381c67f2aadc7f de-dupes the `neuron-vim` entry by removing the unmaintained plugin from the plugin list and generated nix expression.

**Comment by wbadart** [comment]
Conflict arose with the base; just fixed that.

**Review by jpotier — APPROVED** [review]

**Review by jpotier — APPROVED** [review]

**Comment by wbadart** [comment]
Hey team, any further blockers?

/cc @SuperSandro2000 @jonringer

**Review by jonringer — COMMENTED** [review]
do you mind squashing, and rebasing. This should only be one commit

Please follow [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md) and [manual#submitting-changes-making-patches ](https://nixos.org/nixpkgs/manual/#submitting-changes-making-patches) and squash the fix-up commits.

`git rebase -i` is a great tool for this, I created a small video demonstrating it's use [here](https://www.youtube.com/watch?v=OR8Q-4PLalc). A more indepth text tutorial can be found [here](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase)

**Comment by wbadart** [comment]
You got it, thanks Jonathan. I had originally spread it out because I was following the vim plugin-specific [instructions] (using update.py). I just rebased so this should hopefully be more in line with the broader contributing policy!

[instructions]: https://github.com/NixOS/nixpkgs/blob/nixos-unstable/doc/languages-frameworks/vim.section.md#adding-new-plugins-to-nixpkgs

File: pkgs/misc/vim-plugins/generated.nix [hunk 2]
```diff
@@ -2472,14 +2472,14 @@ let
...
     pname = "neuron-vim";
-    version = "2020-08-07";
+    version = "2020-10-29";
     src = fetchFromGitHub {
-      owner = "ihsanturk";
+      owner = "fiatjaf";
       repo = "neuron.vim";
-      rev = "07521a3ef2940bd726e7b4d50b82e46898e686cc";
-      sha256 = "0myadiy6y2p73lhdzk2w55whg4i5rs004jaw1m21cz0dk8k8ibn2";
+      rev = "2b4321bf12a4d0b589cd8c26aabb7e31311dab26";
```
Comment (jpotier) [inline pkgs/misc/vim-plugins/generated.nix:2479]: Could you pick the latest rev (if it's not too much work)? There was a nice fix 2 days ago.

Comment (jpotier) [reply pkgs/misc/vim-plugins/generated.nix:2479]: https://github.com/fiatjaf/neuron.vim/commit/c62b5daa9bc9d734eca9db74e7e0895c85ceb2b8

**Review by jpotier — APPROVED** [review]

File: pkgs/misc/vim-plugins/generated.nix (hunk 2 above)
Comment (jonringer) [reply pkgs/misc/vim-plugins/generated.nix:2479]: these automatically get updated through a script, the newest version should land in master shortly

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 101385 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary>  <ul>    <li>vimPlugins.neuron-vim</li>  </ul></details>

**Comment by primeos** [comment]
@wbadart you somehow managed to make me the commit author in your last force-push... :o xD

**Comment by wbadart** [comment]
Git is truly an enigma sometimes :) Welcome to the thread, I guess!
